addons/hr_attendance/report/attendance_errors.py

```diff
@@ -3,33 +3,25 @@
 
 import datetime
 import time
-from openerp.osv import osv
-from openerp.report import report_sxw
+from openerp import api, models
 
 
-class attendance_print(report_sxw.rml_parse):
-
-    def __init__(self, cr, uid, name, context):
-        super(attendance_print, self).__init__(cr, uid, name, context=context)
-        self.localcontext.update({
-            'time': time,
-            'lst': self._lst,
-            'total': self._lst_total,
-            'get_employees':self._get_employees,
-        })
+class ReportHrAttendanceerrors(models.AbstractModel):
+    _name = 'report.hr_attendance.report_attendanceerrors'
+    _inherit = 'report.abstract_report'
+    _template = 'hr_attendance.report_attendanceerrors'
 
     def _get_employees(self, emp_ids):
-        emp_obj_list = self.pool.get('hr.employee').browse(self.cr, self.uid, emp_ids)
-        return emp_obj_list
+        return self.env['hr.employee'].browse(emp_ids)
 
-    def _lst(self, employee_id, dt_from, dt_to, max, *args):
-        self.cr.execute("select name as date, create_date, action, create_date-name as delay from hr_attendance where employee_id=%s and to_char(name,'YYYY-mm-dd')<=%s and to_char(name,'YYYY-mm-dd')>=%s and action IN (%s,%s) order by name", (employee_id, dt_to, dt_from, 'sign_in', 'sign_out'))
-        res = self.cr.dictfetchall()
+    @api.multi
+    def _lst(self, employee_id, dt_from, dt_to, max):
```

Comment:
_lst is going to be call from report template, then I think there is no need of decoration